File: packages/health_connector_hc_android/android/src/main/kotlin/com/phamtunglam/health_connector_hc_android/mappers/health_measurement_unit_mappers/PowerMapper.kt
package com.phamtunglam.health_connector_hc_android.mappers.health_measurement_unit_mappers

import androidx.health.connect.client.units.Power
import com.phamtunglam.health_connector_hc_android.pigeon.PowerDto

/**
 * Converts a [PowerDto] to a Health Connect [Power] object.
 */
internal fun PowerDto.toHealthConnect(): Power = Power.watts(watts)

/**
 * Converts a Health Connect [Power] object to a [PowerDto].
 *
 * Uses watts as the transfer unit for consistency.
 */
internal fun Power.toDto(): PowerDto = PowerDto(
    watts = inWatts,
)

File: packages/health_connector_hc_android/lib/src/mappers/measurement_unit_mappers/power_mapper.dart
import 'package:health_connector_core/health_connector_core_internal.dart'
    show Power;
import 'package:health_connector_hc_android/src/pigeon/health_connector_hc_android_api.g.dart'
    show PowerDto;
import 'package:meta/meta.dart' show internal;

/// Converts [Power] to [PowerDto].
@internal
extension PowerToDto on Power {
  PowerDto toDto() {
    // Uses watts as the transfer unit for consistency.
    return PowerDto(watts: inWatts);
  }
}

/// Converts [PowerDto] to [Power].
@internal
extension PowerDtoToDomain on PowerDto {
  Power toDomain() {
    return Power.watts(watts);
  }
}
